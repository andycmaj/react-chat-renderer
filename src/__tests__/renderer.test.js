import React from 'react';
import SlackRenderer from '../renderer';

import { Message, Text, DividerBlock, SectionBlock } from '../index';

describe('renderer', () => {
  it('renders Hello, world', () => {
    const message = SlackRenderer.render(
      <Message responseType="in_channel">
        <Text>Hello, world</Text>
      </Message>
    );

    expect(message).toMatchSnapshot();
  });

  it('renders an array of Blocks', () => {
    const message = SlackRenderer.render(
      <Message>
        <DividerBlock />
        <SectionBlock>
          <Text>section text</Text>
        </SectionBlock>
      </Message>
    );

    console.log(JSON.stringify(message));

    expect(message).toMatchSnapshot();
  });
});
