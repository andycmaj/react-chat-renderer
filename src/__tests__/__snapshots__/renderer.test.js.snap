// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer renders Hello, world 1`] = `
Object {
  "response_type": "in_channel",
  "text": "Hello, world",
}
`;

exports[`renderer renders an array of Blocks 1`] = `
Object {
  "blocks": Array [
    Object {
      "type": "divider",
    },
    Object {
      "text": Object {
        "text": "section text",
      },
      "type": "section",
    },
  ],
  "response_type": "ephemeral",
}
`;
